node-phantom.js

```diff
@@ -227,6 +227,11 @@ module.exports = {
         var prematureExitHandler = function (code, signal) {
           console.warn('phantom crash: code ' + code)
           server.close()
+
+          // required to signal up the call stack that phantom has crashed
+          // the catching block will signal a error return code allowing
+          // a supervisor process to restart
+          throw("premature phantom crash");
```

Comment:
You are probably correct that process.exit(2) would work here. I guess the only advantage would be some theoretical handling that might be done by a calling function. But I am simply using it as a signally device.